docs/decisions/0004-external-event-bus-and-django-signal-events.rst

```diff
@@ -0,0 +1,52 @@
+External event bus and Django Signal events
+===========================================
+
+Status
+------
+
+Provisional
+
+Context
+-------
+
+OpenedX services already use Django signals for internal events (`OEP-49: Django App Pattern <https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0049-django-app-patterns.html#signals>`_). Additionally, we are adding the ability to trigger external events using an Event Bus(`OEP-52: Event Bus <https://github.com/openedx/open-edx-proposals/pull/233>`_).
+
+This ADR concerns decisions made during trial of event bus regarding how external events will be triggered, and how external event bus builds off of decisions made for internal events.
+
+This decision came out based on following conversations:
+
+- `#38: What to send over the wire (Kafka)? <https://github.com/eduNEXT/openedx-events/issues/38>`_
+
+- `#39: ARCHBOM-2010: How does Event Bus interact with OpenedX Django Signals? <https://github.com/eduNEXT/openedx-events/issues/39>`_
+
+Decision
+--------
+
+- Event definitions (in the form of OpenEdxPublicSignal) will be shared between internal and external events.
+
+- At this time, event bus events will be triggered from a Django signal handler of the Django signal representing the corresponding internal event.
+
+- The Django signal must contain all event envelope and data to be sent across the event bus, such that it can be converted back into Django signals again on the consumer side.
+
+- For consumption, the event bus implementation will convert the messages back into django signals and emit them within the consumer application.
+
+Consequences
+------------
+
+- The OpenEdxPublicSignal serves as the event definition, and doubles as a Django signal.
+
```

Comment:
Insert:
```
- In other words, the OpenEdxPublicSignal provides the schema for external events.
```
Note: this is useful language to refer back to from the next ADR that digs into these details.